[Zhou GuanZong] ip

Gnoznaug opened the pull request:
# Duke Pro
🔥 

> “Your mind is for having ideas, not holding them.” – David Allen

Some features include
- Adding tasks
- Deleting tasks
- Marking tasks

Example:
```
1. [X] Read book
2. [ ] Return book (by: Feb 05 2022)
```
Thing to add:
- [x] PR description
- [ ] PR review
- [ ] Increments

Link to my repo:
[Here](https://github.com/Gnoznaug/ip)

Main method:
```Java
public static void main(String[] args) {
        new Duke("data/duke.txt").run();
}
```
PR update requirements:
~~1. a heading~~
~~2. a bullet list~~
~~3. a numbered list~~
~~4. a fenced code block (with syntax highlighting)~~
~~5. a task list~~
~~6. an emoji~~
~~7. a blockquote~~
~~8. a hyper link~~
~~9. inline code~~
~~10. some text formatting: bold, italic, stikethrough etc.~~

Commit:
Add Gradle support

Gnoznaug committed:
add greet, echo, exit

Gnoznaug committed:
add Add, List

Gnoznaug committed:
add mark as done

Gnoznaug committed:
add todos, events, deadlines

Gnoznaug committed:
add automated text ui testing

Gnoznaug committed:
add handle errors

Gnoznaug committed:
Add delete feature

Gnoznaug committed:
Add save feature

Gnoznaug committed:
Add time format

Gnoznaug committed:
Fix previous commit for time feature

Gnoznaug committed:
Merge branch 'branch-Level-7'

Gnoznaug committed:
Merge branch 'branch-Level-8'

Gnoznaug committed:
Employ OOP extensively

Gnoznaug committed:
Divide classes into packages

Gnoznaug committed:
Add JUnit tests for Parser and TaskList

Gnoznaug committed:
Add JavaDoc comments

Gnoznaug committed:
Fix code to follow Coding Standard

Gnoznaug committed:
Add functionality to find tasks that contain a given searchTerm

Gnoznaug committed:
Fix JavaDoc comment on MarkCommand constructor

Gnoznaug committed:
Merge branch 'branch-A-JavaDoc'

Gnoznaug committed:
Merge branch 'branch-A-CodingStandard'

Gnoznaug committed:
Merge branch 'branch-Level-9'

Gnoznaug committed:
Merge branch 'add-gradle-support' into A-Gradle

Gnoznaug committed:
Add Gradle support

Gnoznaug committed:
Merge branch 'branch-A-Gradle'

Gnoznaug committed:
Use CheckStyle and fix style violations

Gnoznaug committed:
Merge branch 'branch-A-CheckStyle'

Gnoznaug committed:
Add GUi for Duke

Gnoznaug committed:
Merge branch 'branch-Level-10'

Gnoznaug committed:
Fix coding style violations

iztanpy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+package duke;
+
+import java.io.EOFException;
+
+import duke.command.Command;
+import duke.exception.DukeException;
+import duke.manager.Parser;
+import duke.manager.Storage;
+import duke.manager.TaskList;
```
nicely done on making sure not to use generic import *

iztanpy commented on src/main/java/duke/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,58 @@
+/**
+ * Represents a command that will add a Deadline Task to the TaskList upon execution..
+ */
+public class AddDeadlineCommand extends Command {
```
An alternative could be to have all the commands in one class, but I can understand that this might be neater too! :)

iztanpy commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,62 @@
+        try {
+            storage.save(taskList);
+        } catch (Exception e) {
+            throw new DukeException(e.getMessage());
```
Nice! I think that this is more intuitive, I was previously typing out unique error messages for each error but I think this is a smarter solution!

iztanpy reviewed (commented):
I like the use of classes, and how this project utilised oop to the max! :))

Gnoznaug committed:
Add assert statements around codebase

Assertions are used to define assumptions about the program state.

Helps to verify assumptions about the internal invariants in the code.

Gnoznaug committed:
Fix code to improve code quality

Changed parse() in the Parse class for SLAP, it makes the code more
readable.

Deleted dead code in MainWindow Class to reduce error-prone practices.

Gnoznaug committed:
Merge pull request #1 from Gnoznaug/branch-A-Assertions

Add assert statements around codebase

Gnoznaug committed:
Merge branch 'master' into branch-A-CodeQuality

Gnoznaug committed:
Merge pull request #2 from Gnoznaug/branch-A-CodeQuality

Improve Code Quality

Gnoznaug committed:
Add update task feature

Gnoznaug committed:
Merge pull request #3 from Gnoznaug/branch-C-Update

Add update task feature

Gnoznaug committed:
Fix easy-to-detect problems regarding code quality

Gnoznaug committed:
Add User Guide

Gnoznaug committed:
Set theme jekyll-theme-tactile

Gnoznaug committed:
Set theme jekyll-theme-slate

Gnoznaug committed:
Update README.md

Fix MD format

Gnoznaug committed:
Remove dead print statement in MainWindow
